app/Actions/FulfillTrade.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace App\Actions;
+
+use App\Models\Entry;
+use App\Models\Trade;
+use App\Models\Item;
+use App\Models\Character;
+use App\Models\User;
+use App\Exceptions\TradeException;
+use Illuminate\Support\Facades\Auth;
+use Lorisleiva\Actions\Concerns\AsAction;
+
+class FulfillTrade
+{
+    use AsAction;
+
+    /**
+     * Expect an array of item details, create each item and attach them to the appropriate relation objects
+     *
+     * @param Entry $entry
+     * @param array $items
+     */
+    public function handle(Trade $trade, Item $offeredItem)
+    {
+        $tradeCharacter = $trade->character;
+        $tradeItem = $trade->item;
+        $offerCharacter = $offeredItem->character;
+
+        $currentUser = Auth::user();
+        $offerUser = $offerCharacter->user;
+
+        if (!$currentUser->characters->contains($tradeCharacter) || $trade->status == Trade::STATUS_CLOSED) {
+            throw new TradeException("The trade could not be fulfilled.");
+        }
+
+        $tradeItem->character()->disassociate()->save();
+        $offeredItem->character()->dissociate()->save();
+
+        $offeredItem->character()->associate($tradeCharacter->id)->save();
+        $tradeItem->character()->associate($offerCharacter->id)->save();
+
+        $entryData = [
+            'date_played' => now(),
+            'type' => Entry::TYPE_DOWNTIME,
+            'levels' => 0,
+            'notes' => "Trade between ".$tradeCharacter->name." and ".$offerCharacter->name
+        ];
```

Comment:
This should also subtract 5 downtime days

Reply:
done